File: Java/_345_Reverse_Vowels_of_a_String/Solution.java
class Solution {
    public String reverseVowels(String s) {
        if (s.length() < 1)
            return s;
        int start = 0;
        int end = s.length() - 1;
        char arr[] = s.toCharArray();
        while(start < end) {
            if (checkVowel(arr[start])) {
                if (checkVowel(arr[end])) {
                    char temp = arr[start];
                    arr[start] = arr[end];
                    arr[end] = temp;
                    start++;
                    end--;
                }
                else end--;
            }
            else start++;
        }
        return String.valueOf(arr);
    }
    
    public static boolean checkVowel(char c) {
        if ('a' == c || 'e' == c || 'i' == c || 'o' == c || 'u' == c
                || 'A' == c || 'E' == c || 'I' == c || 'O' == c || 'U' == c) {
            return true;
        }
        return false;
    }
}

File: Python/_345_Reverse_Vowels_of_a_String/solution.py
class Solution:
    def reverseVowels(self, s):
        """
        :type s: str
        :rtype: str
        """
        vowels = {'a', 'e', 'i', 'o', 'u', 'A', 'E', 'I', 'O', 'U'}
        slist = list(s)
        i = 0
        j = len(slist) - 1
        while i < j:
            while i < j and slist[i] not in vowels:
                i += 1
            while j > i and slist[j] not in vowels:
                j -= 1
            slist[i], slist[j] = slist[j], slist[i] 
            i += 1
            j -= 1
        return ''.join(slist)

